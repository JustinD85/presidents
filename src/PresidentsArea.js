import React, { Fragment } from 'react'
import { connect } from 'react-redux'
const styles = {
  root: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  card: {
    margin: 8,
    minWidth: 250,
    background: 'lightgrey',
    padding: 10,
    borderRadius: 10,
  },
  filter: {
    height: 100,
    margin: '0 auto',
  },
  loading: { margin: '0 auto' },
}
export const PresidentsArea = ({ presidents, hasErrored }) => {
  if (hasErrored) {
    return (
      <h1 className="error" style={{ margin: '0 auto' }}>
        There is an error...
      </h1>
    )
  } else if (presidents) {
    const ufilter = presidents.map(p => p.party)
    const list = ufilter.filter(
      (word, index) => ufilter.findIndex(iWord => iWord === word) === index,
    )
    return (
      <div className="presidents" style={styles.root}>
        <div style={styles.filter}>
          <select>
            {list.map(party => (
              <option>{party}</option>
            ))}
          </select>
        </div>
        {presidents.map(president => (
          <Fragment key={president.toString + 'key'}>
            <div style={styles.card} key={president.toString()}>
              <p>Order: {president.number}</p>
              <p>Name: {president.president}</p>
              <p>Birth: {president.birth_year}</p>
              <p>Death: {president.death_year}</p>
              <p>Took Office: {president.took_office}</p>
              <p>Left Office: {president.left_office}</p>
              <p>Affiliation: {president.party}</p>
            </div>
          </Fragment>
        ))}
      </div>
    )
  } else {
    return (
      <h1 className="loading" style={styles.loading}>
        Loading please wait...
      </h1>
    )
  }
}

export const mapStateToProps = state => ({
  presidents: state.presidents,
  isLoading: state.isLoading,
  hasErrored: state.hasErrored,
})
export default connect(mapStateToProps)(PresidentsArea)
