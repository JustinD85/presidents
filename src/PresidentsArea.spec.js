import React from 'react'
import { shallow } from 'enzyme'
import mockData from './mockData'

import { PresidentsArea, mapStateToProps } from './PresidentsArea'
describe('PresidentsArea', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<PresidentsArea />)
  })

  it('should render without error', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should show loading screen if no error or data', () => {
    expect(wrapper.find('.loading').length).toEqual(1)
  })

  it('should show error if there is one ', () => {
    wrapper = shallow(<PresidentsArea hasErrored={true} />)
    expect(wrapper.find('.error').length).toEqual(1)
  })

  it('should show presidents if data', () => {
    wrapper = shallow(<PresidentsArea presidents={mockData} />)
    expect(wrapper.find('.presidents').length).toEqual(1)
  })

  describe('mapStateToProps', () => {
    const mockState = {
      stuff: 'not this',
      presidents: mockData,
      isLoading: false,
      hasErrored: false,
    }
    const expectation = {
      presidents: mockData,
      isLoading: false,
      hasErrored: false,
    }
    const props = mapStateToProps(mockState)
    expect(expectation).toEqual(props)
  })
})
