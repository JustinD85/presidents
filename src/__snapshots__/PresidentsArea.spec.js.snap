// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PresidentsArea should render without error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PresidentsArea />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Loading please wait...",
      "className": "loading",
      "style": Object {
        "margin": "0 auto",
      },
    },
    "ref": null,
    "rendered": "Loading please wait...",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading please wait...",
        "className": "loading",
        "style": Object {
          "margin": "0 auto",
        },
      },
      "ref": null,
      "rendered": "Loading please wait...",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
